scio-core/src/main/scala/com/spotify/scio/values/PairSCollectionFunctions.scala

```diff
@@ -376,13 +417,38 @@ class PairSCollectionFunctions[K, V](val self: SCollection[(K, V)])
    * @param fpProb false positive probability when computing the overlap
    */
   def sparseOuterJoin[W: ClassTag](that: SCollection[(K, W)],
-                                   thatNumKeys: Int,
+                                   thatNumKeys: Long,
                                    fpProb: Double = 0.01)
                                   (implicit hash: Hash128[K])
   : SCollection[(K, (Option[V], Option[W]))] = {
+    val bfSettings = PairSCollectionFunctions.optimalBFSettings(thatNumKeys, fpProb)
+    if (bfSettings.numBFs == 1) {
+      sparseOuterJoinImpl(that, thatNumKeys.toInt, fpProb)
+    } else {
+      val n = bfSettings.numBFs
+      val thisParts = self.partition(n, _._1.hashCode() % n)
+      val thatParts = that.partition(n, _._1.hashCode() % n)
+      val joined = (thisParts zip thatParts).map { case (lhs, rhs) =>
+        lhs.sparseOuterJoinImpl(rhs, bfSettings.capacity, fpProb)
+      }
+      SCollection.unionAll(joined)
+    }
+  }
+
+  protected def sparseOuterJoinImpl[W: ClassTag](that: SCollection[(K, W)],
+                                               thatNumKeys: Int,
+                                               fpProb: Double)
+                                              (implicit hash: Hash128[K])
+  : SCollection[(K, (Option[V], Option[W]))] = {
     val width = BloomFilter.optimalWidth(thatNumKeys, fpProb).get
     val numHashes = BloomFilter.optimalNumHashes(thatNumKeys, width)
-    val rhsBf = that.keys.aggregate(BloomFilterAggregator[K](numHashes, width)).asSingletonSideInput
+    val rhsBf = {
```

Comment:
You prefer this over the `skewedJoin` solution, which uses `asIterableSideInput` and then when used `if (c(rhsBf).nonEmpty && c(rhsBf).head.maybeContains(e._1)) {`?